cruise-control/src/test/java/com/linkedin/kafka/cruisecontrol/servlet/AuthenticationIntegrationTest.java

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright 2017 LinkedIn Corp. Licensed under the BSD 2-Clause License (the "License"). See License in the project root for license information.
+ */
+
+package com.linkedin.kafka.cruisecontrol.servlet;
+
+import com.linkedin.kafka.cruisecontrol.CruiseControlIntegrationTestHarness;
+import com.linkedin.kafka.cruisecontrol.config.KafkaCruiseControlConfig;
+import org.apache.http.auth.BasicUserPrincipal;
+import org.eclipse.jetty.security.Authenticator;
+import org.eclipse.jetty.security.ConstraintMapping;
+import org.eclipse.jetty.security.DefaultIdentityService;
+import org.eclipse.jetty.security.DefaultUserIdentity;
+import org.eclipse.jetty.security.IdentityService;
+import org.eclipse.jetty.security.LoginService;
+import org.eclipse.jetty.security.authentication.BasicAuthenticator;
+import org.eclipse.jetty.server.UserIdentity;
+import org.eclipse.jetty.util.security.Constraint;
+import org.eclipse.jetty.util.security.Credential;
+import org.junit.Test;
+
+import javax.security.auth.Subject;
+import javax.servlet.ServletRequest;
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import static org.junit.Assert.assertEquals;
+
+public class AuthenticationIntegrationTest extends CruiseControlIntegrationTestHarness {
+
+  private static final String TEST_USER = "test";
+  private static final String TEST_PASSWORD = "12345";
+
+  @Override
+  protected Map<String, Object> withConfigs() {
+    return Collections.singletonMap(KafkaCruiseControlConfig.SECURITY_PROVIDER, BasicSecurityProvider.class.getName());
+  }
+
+  @Test
+  public void testSuccessfulAuthentication() throws IOException, URISyntaxException {
+    HttpURLConnection connection = (HttpURLConnection) new URI(_app.serverUrl())
+        .resolve("kafkacruisecontrol/state").toURL().openConnection();
+    String encoded = Base64.getEncoder().encodeToString((TEST_USER + ":" + TEST_PASSWORD).getBytes(StandardCharsets.UTF_8));
+    connection.setRequestProperty("Authorization", "Basic " + encoded);
+    assertEquals(200, connection.getResponseCode());
```

Comment:
Nit: Is it possible to use status codes from the `HttpServletResponse` (e.g. `javax.servlet.http.HttpServletResponse.SC_OK`) here and in the rest of the test?